pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -11447,78 +11447,207 @@ spec:
 
 	testCases := []struct {
-		name        string
-		pipelinerun []*v1beta1.PipelineRun
-		pipeline    []*v1beta1.Pipeline
-		task        []*v1beta1.Task
+		name          string
+		pipelinerun   []*v1beta1.PipelineRun
+		pipelineBytes []byte
+		taskBytes     []byte
+		pipelineError bool
 	}{
 		{
-			name:        "unsigned pipeline fails verification",
-			pipelinerun: []*v1beta1.PipelineRun{prs},
-			pipeline:    []*v1beta1.Pipeline{ps},
+			name:          "unsigned pipeline fails verification",
+			pipelineBytes: unsignedPipelineBytes,
+			taskBytes:     unsignedTaskBytes,
+			pipelineError: true,
 		},
 		{
-			name:        "signed pipeline with unsigned task fails verification",
-			pipelinerun: []*v1beta1.PipelineRun{prs},
-			pipeline:    []*v1beta1.Pipeline{signedPipeline},
-			task:        []*v1beta1.Task{ts},
+			name:          "signed pipeline with unsigned task fails verification",
+			pipelineBytes: signedPipelineWithUnsignedTaskBytes,
+			taskBytes:     unsignedTaskBytes,
+			pipelineError: false,
 		},
 		{
-			name:        "signed pipeline with modified task fails verification",
-			pipelinerun: []*v1beta1.PipelineRun{prs},
-			pipeline:    []*v1beta1.Pipeline{signedPipeline},
-			task:        []*v1beta1.Task{tamperedTask},
+			name:          "signed pipeline with modified task fails verification",
+			pipelineBytes: signedPipelineWithModifiedTaskBytes,
+			taskBytes:     modifiedTaskBytes,
+			pipelineError: false,
 		},
 		{
-			name:        "modified pipeline with signed task fails verification",
-			pipelinerun: []*v1beta1.PipelineRun{prs},
-			pipeline:    []*v1beta1.Pipeline{tamperedPipeline},
-			task:        []*v1beta1.Task{signedTask},
+			name:          "modified pipeline with signed task fails verification",
+			pipelineBytes: modifiedPipelineBytes,
+			taskBytes:     signedTaskBytes,
+			pipelineError: true,
 		},
 	}
-
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			prs := parse.MustParseV1beta1PipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: test-pipelinerun
+  namespace: foo
+  selfLink: /pipeline/1234
+spec:
+  pipelineRef:
+    resolver: %s
+  serviceAccountName: default
+`, resolverName))
+
 			d := test.Data{
-				PipelineRuns:         tc.pipelinerun,
-				Pipelines:            tc.pipeline,
-				Tasks:                tc.task,
+				PipelineRuns:         []*v1beta1.PipelineRun{prs},
+				ConfigMaps:           cms,
 				VerificationPolicies: vps,
 			}
 			prt := newPipelineRunTest(t, d)
 			defer prt.Cancel()
 
-			reconciledRun, _ := prt.reconcileRun("foo", "test-pipelinerun", []string{}, true)
-
+			reconciledRun, _ := prt.reconcileRun("foo", "test-pipelinerun", []string{}, false)
+			// reconcile to resolve the resolution request for pipeline, and create the resolution request for task if pipeline resolved
+			reconciledRun = resolveResolutionRequestandReconcile(t, prt, resolverName, tc.pipelineBytes, "foo", "test-pipelinerun", tc.pipelineError)
```

Comment:
I mentioned this in another comment but it would be cleaner if possible to seed resolution requests in the test data rather than trying to run through multiple reconcile loops.

Reply:
Sure! The latest commit should have adopted this suggestion